tests/php/View/HTMLTest.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace SilverStripe\View\Tests;
+
+use InvalidArgumentException;
+use SilverStripe\Dev\SapphireTest;
+use SilverStripe\View\HTML;
+
+class HTMLTest extends SapphireTest
+{
+    public function testCreateVoidTag()
+    {
+        $tag = HTML::createTag('meta', [
+            'name' => 'description',
+            'content' => 'test tag',
+        ]);
+        $this->assertEquals('<meta name="description" content="test tag" />', $tag);
+    }
+
+    public function testNormalTag()
+    {
+        $tag = HTML::createTag('a', [
+            'title' => 'Some link',
+            'nullattr' => null,
+        ]);
+        $this->assertEquals('<a title="Some link"></a>', $tag);
+
+        $tag = HTML::createTag('a', [
+            'title' => 'HTML & Text',
+            'nullattr' => null,
+        ], 'Some <strong>content!</strong>');
+        $this->assertEquals('<a title="HTML &amp; Text">Some <strong>content!</strong></a>', $tag);
+    }
+
+    public function testVoidContentError()
+    {
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage("Void element \"link\" cannot have content");
+        $tag = HTML::createTag('link', [
+            'title' => 'HTML & Text',
+            'nullattr' => null,
+        ], 'Some <strong>content!</strong>');
+        $this->assertEquals('<a title="HTML &amp; Text">Some <strong>content!</strong></a>', $tag);
```

Comment:
I'm confused with why it's asserting `'<a title="HTML &amp; Text">Some <strong>content!</strong></a>'`, does this assert even trigger?
if that's the case, perhaps add a comment above to state that.

Although I've noticed another `expectException()` below and doesn't have any asserts following the code which throws an exception